### ISSUE TITLE
No way to assign users to roles?

### ISSUE THREAD

**Issue by Monsterhulk** [spatie/laravel-permission#727]
Sorry, 
maybe this has been asked before but it seems there is no way to assign users to roles cause of your polymorphic relations. In my laravel application i have to assign users to roles and roles to users. I have tried it with sync, attach, save and saveMany but nothing seems to work. Have i overseen something in the docs or is it even impossible to assign an array of users/user id's to a single role?

**Comment by drbyte** [comment]
You're right: there's no supplied method to directly sync a group of users to a specific role. Users is just one of many models that roles can be attached to.

However, you can sync roles to users using `$user->syncRoles()`
https://github.com/spatie/laravel-permission/blob/master/src/Traits/HasRoles.php#L110-L122

For mass-assigning users to roles you can use that same method with a slight adjustment to your posted code:
```php
// assumes $users is an array of integer ids; This could be altered to handle objects as well.
foreach($users as $user) {
  if ($user = User::find($user)) {
    $user->syncRoles($role->id);
  }
}
```
You could similarly use `each()` on a collection.

The benefit of using the supplied methods is that it also keeps the cache fresh at the same time.

**Comment by edmundss** [comment]
Try this:

`$role = App\Models\Role::where('name', 'vision-user')->first();
$user_ids = App\User::where('organization_id', 126)->pluck('id');

return $role->users()->sync($user_ids);`

Output:

`{
attached: [
510,
517,
575,
576,
577
],
detached: [ ],
updated: [ ]
}`
